waku/node/v2/waku_types.nim

```diff
@@ -146,8 +148,21 @@ type
     #multiaddrStrings*: seq[string]
 
   WakuResult*[T] = Result[T, cstring]
-# Encoding and decoding -------------------------------------------------------
 
+  Beneficiary* = seq[byte]
+
+  # TODO Consider adding payment threshhold and terms field
+  Handshake* = object
+    beneficiary*: Beneficiary
+
+  Cheque* = object
+    beneficiary*: Beneficiary
+    date*: uint32
+    amount*: uint32
```

Comment:
For the sake of authentication, a signature field can be added to the Cheque.

Reply:
Yep, I'll do that in follow up PRs. Still thinking about how this should work exactly